src/android.ts

```diff
@@ -162,11 +144,29 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
             );
             if (!fridaIsStarted) throw new Error('Failed to start Frida.');
         },
+        async hasDeviceBooted(options) {
+            const waitForDevice = options?.waitForDevice ?? false;
+            const { stdout: devBootcomplete } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'dev.bootcomplete'],
+                { reject: false, timeout: 200 }
+            );
+            const { stdout: sysBootCompleted } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'sys.boot_completed'],
+                { reject: false, timeout: 200 }
+            );
+            const { stdout: bootanim } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'init.svc.bootanim'],
+                { reject: false, timeout: 200 }
+            );
+            return devBootcomplete.includes('1') && sysBootCompleted.includes('1') && bootanim.includes('stopped');
+        },
         async requireRoot(action) {
             if (!options.capabilities.includes('root')) throw new Error(`Root access is required for ${action}.`);
 
             await adb(['root']);
-            await this.awaitAdb();
+            await adb(['wait-for-device'], { timeout: 400 }).catch((e) => {
```

Comment:
400ms doesn't seem like enough to me. We might well be running an emulator in a slow CI or whatever. Let's make that at least 2.5s.